### ISSUE TITLE
A question to virtual environment

### ISSUE THREAD

**Issue by route2021** [holzschu/Carnets#230]
I have no problem to generate and activate a virtual environment in order to install a pure python package (e.g. psychrolib) in the new environment. With "!pip list" I can identify the newly installed package.
But when I create a new script and put "%run myvenv/bin/activate.py" in the very first line and try to import psychrolib, I get ModuleNotFoundError: No module named 'psychrolib' !

**Comment by holzschu** [comment]
Thanks for raising this issue! I'm going to have a look.

**Comment by holzschu** [comment]
Yep, I forgot to edit `sys.path` in `activate.py`. The next version will have the fix.

**Comment by holzschu** [comment]
The fix is out for Carnets, not yet for Carnets-with-scipy.

**Comment by route2021** [comment]
There is one more problem:
The 'asyncssh' package has a dependency on cryptography, which is part of carnets.  Therefore asyncssh can easily be installed with pip in the default directory of carnets and works very well.  However, if asyncssh is installed in a virtual environment, cryptography can't be found. Other packages with dependencies on libraries built into carnets may not run under venv, I didn't test so.

**Comment by holzschu** [comment]
Indeed, and that's very strange.

**Comment by holzschu** [comment]
Found it. After editing sys.path, I also edited sys.prefix, which is a bad idea.

**Comment by holzschu** [comment]
Thanks for your help. Version 1.6.6 is now out in the AppStore (both apps), and seems to work on my machines.

**Comment by route2021** [comment]
On my machine too.
Thanks a lot!

**Comment by route2021** [comment]
On line 16 of activate.py I found: **del os.environ['VIRTUAL_ENV']**
This causes an error on next line!

I think the order must be changed!? :
virtualEnvPath = os.path.join(os.environ['VIRTUAL_ENV'], "lib/python3.9/site-packages")
sys.path.remove(virtualEnvPath)
**del os.environ['VIRTUAL_ENV']**

**Comment by holzschu** [comment]
You're absolutely right. It will be fixed on the next version. I may have trouble accessing the network for the next few days, but I'll upload the next version on the 26th at the latest.
